tools/releaseBuild/azureDevOps/templates/nuget.yml

```
@@ -182,6 +190,43 @@ jobs:
       Get-ChildItem "$(System.ArtifactsDirectory)\signed\globaltool" -Recurse
     displayName: Move global tool packages to subfolder and capture
 
+  - pwsh: |
+      $packagePath = (Join-Path $(System.ArtifactsDirectory) checksum)
+      New-Item $packagePath -ItemType Directory -Force > $null
+      $srcPaths = @("$(System.ArtifactsDirectory)\finalResults", "$(System.ArtifactsDirectory)\macosPkgResults", "$(System.ArtifactsDirectory)\signed")
+
+      $packages  = Get-ChildItem -Path $srcPaths -Include *.zip, *.tar.gz, *.msi*, *.pkg, *.deb, *.rpm -Exclude "PowerShell-Symbols*" -Recurse
+      $packages | ForEach-Object { Copy-Item $_.FullName -Destination $packagePath -Verbose }
+
+      $message = Get-ChildItem $packagePath -Recurse | Select-Object -ExpandProperty FullName | Out-String
+      Write-Verbose -Verbose $message
+
+      $hashFileContent = [System.Text.StringBuilder]::new()
+
+      Get-ChildItem -Path $packagePath | ForEach-Object {
+        Write-Verbose -Verbose "Generating checksum file for $_"
+        $packageName = Split-Path $_ -Leaf
+        $hash = (Get-FileHash -Path $_ -Algorithm SHA512).Hash.ToLower()
+
+        # the '*' before the packagename signifies it is a binary
+        $hashLine = "$hash *$packageName"
+        $hashFileContent.AppendLine($hashLine) > $null
+      }
+
+      $hashFileContent.ToString() | Out-File -FilePath "$packagePath\SHA512SUMS" -Force
+
+      $fileContent = Get-Content -Path "$packagePath\SHA512SUMS" -Raw | Out-String
```

I don't think `Out-String` is needed here

```suggestion
      $fileContent = Get-Content -Path "$packagePath\SHA512SUMS" -Raw
```

---

@adityapatwardhan Did you want to comment about why this is resolved without applying it?